basic-2/src/test/java/Car.java

```diff
@@ -0,0 +1,31 @@
+
+public class Car extends Vehicle {
+	private int odometer=0;
+	public String model;
+	public boolean isStart;
+	
+	Car(IEngine Iengine){
+		Iengine=new SmallEngine();
+	}
+	
+	Car(String model, IEngine Iengine){
+		 this.model=model;
+		 Iengine=new LargeEngine();
+	}
+	
+	public int getOdometer() {
+		return this.odometer;
+	}
+	
+	public void drive(int distance) {
+		this.odometer+=distance;
+	}
+	
+	public void start(IEngine IEngine) {

```

Comment:
The engine should be included in the ctor but not here since the Car will only have one engine.